src/router.rs

```diff
@@ -195,13 +197,29 @@ impl<'a, Data> Resource<'a, Data> {
             let new_resource = ResourceData {
                 endpoints: HashMap::new(),
                 middleware: self.middleware_base.clone(),
+                method_options: Vec::new(),
             };
             *resource = Some(new_resource);
         }
         let resource = resource.as_mut().unwrap();
+        if method != http::Method::OPTIONS {
+            // It's important that this panics here, otherwise we will get weird results for OPTIONS
+            // routes
+            if resource.endpoints.contains_key(&method) {
+                panic!("A {} endpoint already exists for this path", method)
+            }
 
-        if resource.endpoints.contains_key(&method) {
-            panic!("A {} endpoint already exists for this path", method)
+            resource.method_options.push(method.as_str().to_string());
+            let option_header = resource.method_options.join(", ");
+            self.options(async || {
+                http::Response::builder()
+                    .status(http::status::StatusCode::OK)
+                    .header("Access-Control-Allow-Methods", option_header)
```

Comment:
I think CORS headers would better processed in the dedicated middleware. The middleware might provide additional features, such as allowing specific referrers.

Reply:
Yep that is fair! I'll remove that for now. Once we write a CORS middleware we can incorporate that header into the response